crates/spicepod/src/component/dataset.rs

```diff
@@ -135,6 +135,9 @@ pub struct Dataset {
 
     #[serde(default, skip_serializing_if = "is_default")]
     pub ready_state: ReadyState,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub metrics: Option<Metrics>,
```

Comment:
This is where I added the new metrics definition to the Spicepod dataset definition. I also added it to several other components (i.e. `catalog`, `model`, etc). But those won't be hooked up yet in the runtime.